bagel/tests/test_utility.py

```diff
@@ -414,34 +440,39 @@ def test_get_bids_subjects_simple(bids_path, bids_dir):
 
 
 @pytest.mark.parametrize(
-    "bids_list, expectation",
+    "bids_list, missing_subs",
     [
-        (["sub-01", "sub-02", "sub-03"], does_not_raise()),
+        (["sub-01", "sub-02", "sub-03"], []),
         (
             ["sub-01", "sub-02", "sub-03", "sub-04", "sub-05"],
-            pytest.raises(LookupError),
+            ["sub-04", "sub-05"],
         ),
         (
             ["sub-cbm001", "sub-cbm002", "sub-cbm003"],
-            pytest.raises(LookupError),
+            ["sub-cbm001", "sub-cbm002", "sub-cbm003"],
         ),
         (
             ["sub-pd123", "sub-pd234"],
-            pytest.raises(LookupError),
+            ["sub-pd123", "sub-pd234"],
         ),
     ],
 )
-def test_check_unique_bids_subjects_err(bids_list, expectation):
+def test_get_subjects_missing_from_pheno_data(bids_list, missing_subs):
     """
-    Given a list of BIDS subject IDs, raise an error or not depending on
-    whether all IDs are found in the phenotypic subject list.
+    Given a list of BIDS subject IDs, test that IDs not found in the phenotypic subject list are returned.
     """
     pheno_list = ["sub-01", "sub-02", "sub-03", "sub-PD123", "sub-PD234"]
 
-    with expectation:
-        butil.check_unique_bids_subjects(
-            pheno_subjects=pheno_list, bids_subjects=bids_list
+    # We sort the list for comparison since the order of the missing subjects is not guaranteed
+    # due to using set operations
+    assert (
+        sorted(
+            get_subs_missing_from_pheno_data(
+                pheno_subjects=pheno_list, subjects=bids_list
+            )
         )
+        == missing_subs
+    )
 
```

Comment:
Let's stick with the classic test structure of: setup, act, assert. Better for readability and for making changes later.